src/rard/research/tests/views/test_fragment.py

```diff
@@ -492,3 +511,131 @@ def test_compare_orders(self):
             response.context_data["object_list"],
             [self.aftl1, self.aftl3, self.aftl2, self.aftl3],
         )
+
+
+class TestFragmentDuplicationView(TestCase):
+    def setUp(self):
+        self.cw = CitingWork.objects.create(title="citing work title")
+        self.ot = OriginalText.objects.create(
+            content="Original Text test",
+            apparatus_criticus_blank=False,
+            apparatus_criticus="here be the app crit",
+            object_id=111,
+            citing_work=self.cw,
+            content_type_id=12,
+            reference_order="reference order",
+        )
+        self.ot2 = OriginalText.objects.create(
+            content="Original Text2 test",
+            apparatus_criticus_blank=False,
+            apparatus_criticus="here be the app crit",
+            object_id=121,
+            citing_work=self.cw,
+            content_type_id=14,
+            reference_order="reference order",
+        )
+        self.con = Concordance.objects.create(
+            original_text=self.ot, source="tester", identifier="123"
+        )
+        self.con = Concordance.objects.create(
+            original_text=self.ot2, source="tester", identifier="123"
+        )
+        self.apc = ApparatusCriticusItem.objects.create(
+            parent=self.ot, content="critical test", object_id=23
+        )
+        self.apc2 = ApparatusCriticusItem.objects.create(
+            parent=self.ot2, content="critical test", object_id=23
+        )
+        self.tr = Translation.objects.create(
+            translated_text="translation of text", original_text=self.ot
+        )
+        self.tr2 = Translation.objects.create(
+            translated_text="translation of text", original_text=self.ot2
+        )
+        self.ref = Reference.objects.create(editor="test", original_text=self.ot)
+        self.ref2 = Reference.objects.create(editor="test", original_text=self.ot2)
+        self.topic = Topic.objects.create(name="topic1")
+        self.frag = Fragment.objects.create(name="test fragment")
+        self.frag.topics.add(self.topic)
+        self.frag.original_texts.add(self.ot)
+        self.anonfrag = AnonymousFragment.objects.create(name="test anonfragment")
+        self.anonfrag.topics.add(self.topic)
+        self.anonfrag.original_texts.add(self.ot2)
+
+    def compare_model_objects(self, original, duplicate):
+        for field in original._meta.fields:
+            if field.name in [
+                "id",
+                "created",
+                "modified",
+                "commentary",
+                "object_id",
+                "original_text",
+                "order",
+                "model",
+            ]:
+                continue
+            if field.is_relation and getattr(original, field.name) is not None:
```

Comment:
Why not even if it is none?

Reply:
think it threw an error but may have just been copied over